app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -128,6 +129,12 @@ public void setupUI() {
             activity.triggerUserFormComplete();
         });
 
+        if (LocalePreferences.isLocaleRTL()) {
+            LayoutDirectionUtilCompat.mirrorView(finishButton.findViewById(R.id.nav_image_finish), true);
```

Comment:
Since `nav_image_finish` is a textview, my assumption is it should be getting mirrored by itself without explicitly calling `mirrorView`

Reply:
Makse sense 👍 

I think a better way to mirror views would be in xml as - 

Add android:rotationY="@integer/locale_mirror_flip" to the View you want to be mirrored. And define

res/values/integers.xml
<integer name="locale_mirror_flip">0</integer>

res/values-ldrtl/integers.xml
<integer name="locale_mirror_flip">180</integer>

This will keep the mirroring code in xml. What do you think ?

Reply:
Since layout direction is now being laid-out based on system settings, I'll move this to XML.